static/gcc.js

```diff
@@ -292,11 +293,22 @@ function loadState(state) {
     setFilterUi(state.filterAsm);
     for (var i = 0; i < Math.min(allCompilers.length, state.compilers.length); i++) {
         allCompilers[i].setFilters(state.filterAsm);
-        allCompilers[i].deserialiseState(state.compilers[i]);
+        allCompilers[i].deserialiseState(state.compilers[i],OPTIONS.compilers, OPTIONS.defaultCompiler);
     }
     return true;
 }
 
+function resizeEditors() {
```

Comment:
This code is probably the source of the funny behaviour when the window is resized. It's designed to dynamically calculate the height of the editors to leave enough room at the bottom for the output pane.

As written here, it's not going to work with multiple code mirrors: the "top" on line 303 makes no sense if there's more than one CodeMirror (which there now is :))

This will require a lot more thught but I think we should discuss the UI layout further before embarking on any changes to this.